mmconfig

```diff
@@ -2,416 +2,160 @@
 # mmconfig
 # set up variables for microservices, has option for a graphical user interface
 CONFIG="Y"
-CONFIG_VERSION="1.0"
 SCRIPTDIR=$(dirname "${0}")
 . "${SCRIPTDIR}/mmfunctions" || { echo "Missing '${SCRIPTDIR}/mmfunctions'. Exiting." ; exit 1 ;};
 
-if [[ "$(uname -s)" = "Darwin" ]] ; then
-    RUNPASHUA="Y"
-fi
-
-_usage(){
-    cat <<EOS
-$(basename "${0}")
-  -a  set configuration file using the GUI (default on macOS)
-  -t  set configuration file using the CLI
-EOS
+funcrdoCreate() {
+  RDO_VAR_NAME="${1}"
+  RDO_NAME="${2}"
+  cat <<GTK_RAD
+<label>${RDO_NAME}</label>
+  <variable>${RDO_VAR_NAME}</variable>
+</radiobutton>
+GTK_RAD
 }
 
-# local variables
-REQUIRED_VARIABLES=("OUTDIR_INGESTFILE" "OUTDIR_INGESTXDCAM" "OUTDIR_PAPER" "AIP_STORAGE" "PODCASTDELIVER" "YOUTUBEDELIVER" "TMPDIR" "REGEX4PODCAST" "REGEX4IA" "DVDLABELPREFIX" "OMNEONIP" "OMNEONPATH" "CUSTOM_LOG_DIR" "LTO_INDEX_DIR" "LOCAL_MM_DIR" "EMAIL_FROM" "MAKEYOUTUBE_DELIVERY_EMAIL_TO" "MAKEBROADCAST_DELIVERY_EMAIL_TO" "FILE_MAKER_DB" "VOLADJUST" "PREMIS_DB" "SQL_ERROR_EMAIL_TO" "PREMIS_PROFILE" "PREMIS_NAME" "PERCEPTUAL_HASH")
-
-OPTIND=1
-
-while getopts ":ath" OPT ; do
-    case "${OPT}" in
-        a) RUNPASHUA="Y" ;;
-        t) RUNPASHUA="N" ;;
-        h) _usage ; exit 0 ;;
-        *) echo "bad option -${OPTARG}" ; _usage ; exit 1 ;;
-    esac
-done
-
-_config_edit(){
-    [ -z "${1}" -o -z "${2}" ] && { _report -w "The config_edit function requires two arguments. Error." ; exit ;};
-    SEDTEMP=$(_maketemp)
-    sed "s:^${1}=[^ ]*:$1=$2:" "${MM_CONFIG_FILE}" > "${SEDTEMP}"
-    cp "${SEDTEMP}" "${MM_CONFIG_FILE}"
-}
-
-_test_config(){
-    for DIRECTORYVARIABLE in OUTDIR_INGESTFILE OUTDIR_INGESTXDCAM OUTDIR_PAPER AIP_STORAGE PODCASTDELIVER YOUTUBEDELIVER TMPDIR CUSTOM_LOG_DIR LTO_INDEX_DIR ; do
-        if [ ! -d "${!DIRECTORYVARIABLE}" ] ; then
-            _report -w "${DIRECTORYVARIABLE} is NOT a valid directory"
-        fi
-    done
-    for BOOLEANVARIABLE in VOLADJUST ; do
-        if [[ ! "${!BOOLEANVARIABLE}" = "Y" && ! "${!BOOLEANVARIABLE}" = "N" ]] ; then
-            _report -w "${BOOLEANVARIABLE} is NOT set to Y or N"
-        fi
-    done
-}
-
-_add_key(){
-    KEY_TO_ADD="${1}"
-    grep -q "^${KEY_TO_ADD}=" "${MM_CONFIG_FILE}" ; GREP_CODE="$?"
-    if [[ ! "${GREP_CODE}" -eq "0" ]] ; then
-        _report -td "Adding NEW variable ${KEY_TO_ADD} to ${MM_CONFIG_FILE}"
-        echo "${KEY_TO_ADD}=" >> "${MM_CONFIG_FILE}"
+funcDirectory() {
+    DIR_VAR_NAME="${1}"
+    DIR_VAR_VALUE="${!1}"
+    if [[ -n "${DIR_VAR_VALUE}" ]] ; then 
+      DIR_DEFAULT="<default>${DIR_VAR_VALUE}</default>"
     fi
+    cat << GTK_PART
+<frame ${DIR_VAR_NAME}>
+    <hbox>
+        <entry accept="directory" fs-entry="Select a Directory">
```

Comment:
Can these be arranged so that they are wider? Currently they are so narrow that the full path is rarely seen, so I have to click in the box to move to the end to see what the path is.